### PR TITLE
Fix failing tests in Gutenberg stable and nightly tests

### LINKED ISSUE #48854: [Flaky] customize-store/assembler/homepage.spec.js
Selected homepage should be applied on the frontend 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/9667085189

### LINKED ISSUE #49144: [Flaky] merchant/create-woocommerce-patterns.spec.js
can insert WooCommerce patterns into page 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/9792031534

### LINKED ISSUE #49147: [Flaky] merchant/create-post.spec.js
can create new post 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/9792031534

### LINKED ISSUE #49165: [Flaky] merchant/create-woocommerce-blocks.spec.js
can insert all WooCommerce blocks into page 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/9797263082

### LINKED ISSUE #49178: [Flaky] merchant/create-page.spec.js
can create new page 

Failed on: https://github.com/woocommerce/woocommerce/actions/runs/9802671813

### PR DESCRIPTION
**Pull request by veljkho**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #49178
Closes #49165
Closes #49147
Closes #49144
Closes #48854

This should make daily tests back to green. 🟢 It should be failing Gutenberg stable & nightly tests.
There was a bit of a different UX when having Gutenberg installed, and different selectors for the CYS as well.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

```
pnpm test:e2e-pw tests/e2e-pw/tests/merchant/create-page.spec.js
```
```
pnpm test:e2e-pw tests/e2e-pw/tests/merchant/create-post.spec.js
```
```
pnpm test:e2e-pw tests/e2e-pw/tests/merchant/create-woocommerce-blocks.spec.js
```
```
pnpm test:e2e-pw tests/e2e-pw/tests/merchant/create-woocommerce-patterns.spec.js
```
```
pnpm test:e2e-pw tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js
```

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js [hunk 1]
```diff
@@ -163,21 +164,64 @@ test.describe( 'Assembler -> Homepage', { tag: '@gutenberg' }, () => {
...
+				failOnStatusCode: true,
+			}
+		);
+		const pluginsList = await listPluginsResponse.json();
+		const gutenbergPlugin = pluginsList.find(
+			( { textdomain } ) => textdomain === 'gutenberg'
+		);
+
+		// eslint-disable-next-line playwright/no-conditional-in-test
+		if ( gutenbergPlugin ) {
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: I'm not seeing any difference between the two if branches, other than the snapshot name. Can we simplify it a bit, it's too verbose and hard to follow and understand IMO?

Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: Something like:

```
name: `${ gutenbergPlugin ? 'gutenberg' : ''}-selected-homepage-blocks-class-frontend-${ index }`
```

Comment (rrennick) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: @adimoldovan That only works if the plugin is installed in the `gutenberg` folder. The `textdomain` filter will find it regardless of what folder it's installed in.

Comment (veljkho) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: I can add a comment about it for better clarification.

Comment (veljkho) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: Now is commented and bit rewritten for better understanding.

Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: > @adimoldovan That only works if the plugin is installed in the gutenberg folder. The textdomain filter will find it regardless of what folder it's installed in.

@rrennick I'm not sure what you mean. My suggestion was not about detecting the gutenberg install. I was suggesting another way of using the value of `gutenbergPlugin` which was already set.

Comment (adimoldovan) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: @veljkho The code that gets all the content between the header and the footer is identical in both `if` and `else` branches.
I was suggesting using the ternary operator in the snapshot name instead to keep is shorter.

So instead of 
```
if (withGutenbergPlugin) {
// Get all the content between the header and the footer.
[...]
await expect(...)
.toMatchSnapshot( {
					name: `gutenberg-selected-homepage-blocks-class-frontend-${ index }`,
				}
}
else {
// Get all the content between the header and the footer.
[...]
await expect(...)
.toMatchSnapshot( {
					name: `selected-homepage-blocks-class-frontend-${ index }`,
				}
}
```

use 

```
// Get all the content between the header and the footer.
[...]
await expect(...)
.toMatchSnapshot( {
					name: `${ withGutenbergPlugin ? 'gutenberg' : '' }-selected-homepage-blocks-class-frontend-${ index }`,
				}
}
```

**Review by adimoldovan — CHANGES_REQUESTED** [review]
@veljkho, `create-woocommerce-blocks.spec.js`, `create-woocommerce-patterns.spec.js` and `create-page.spec.js` are failing for me locally with Gutenberg stable. The click on 'Add title' is being blocked by the overlay.

```
TimeoutError: locator.click: Timeout 20000ms exceeded.
Call log:
  - waiting for getByLabel('Add title')
  -   locator resolved to <h1 role="textbox" aria-multiline="true" contenteditable="true" aria-label="Add title" class="wp-block wp-block-post-title block-editor-block-list__block editor-post-title editor-post-title__input rich-text">…</h1>
  - attempting click action
  -   waiting for element to be visible, enabled and stable
  -   element is visible, enabled and stable
  -   scrolling into view if needed
  -   done scrolling
  -   <div class="block-editor-block-preview__container">…</div> from <div class="components-modal__screen-overlay">…</div> subtree intercepts pointer events
```

Playwright Test Report Screenshot

File: plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js (hunk 1 above)
Comment (veljkho) [reply plugins/woocommerce/tests/e2e-pw/tests/customize-store/assembler/homepage.spec.js]: That's a great improvement @adimoldovan , thanks!

**Comment by veljkho** [comment]
Good catch, @adimoldovan I will fix that.

**Comment by veljkho** [comment]
All should be good now @adimoldovan 
I didn't find any way to disable it, instead, I updated `closeWelcomeModal` that we don't use anymore with `closeChoosePatternModal`.

**Comment by adimoldovan** [comment]
> All should be good now @adimoldovan I didn't find any way to disable it, instead, I updated `closeWelcomeModal` that we don't use anymore with `closeChoosePatternModal`.

@veljkho You should use a locator handler instead of waiting for 5s. See https://playwright.dev/docs/api/class-page#page-add-locator-handler

**Comment by veljkho** [comment]
> You should use a locator handler instead of waiting for 5s. See https://playwright.dev/docs/api/class-page#page-add-locator-handler

Wow, that's a cool feature by PW. Thanks!

File: plugins/woocommerce/tests/e2e-pw/utils/editor.js [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
-const closeWelcomeModal = async ( { page } ) => {
-	// Close welcome popup if prompted
-	try {
-		await page
-			.getByLabel( 'Close', { exact: true } )
-			.click( { timeout: 5000 } );
-	} catch ( error ) {
-		// Welcome modal wasn't present, skipping action.
-	}
+const closeChoosePatternModal = async ( { page } ) => {
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/utils/editor.js:3]: I'm not sure the new locator handler code qualifies to be a util function, but I'm ok either way.

**Review by adimoldovan — CHANGES_REQUESTED** [review]
@veljkho I'm still not getting a successful run with Gutenberg. Are you running the tests locally with Gutenberg installed?

```
2 failed
    [Gutenberg] › merchant/create-woocommerce-blocks.spec.js:144:3 › Add WooCommerce Blocks Into Page › can insert all WooCommerce blocks into page 
    [Gutenberg] › merchant/create-woocommerce-patterns.spec.js:32:3 › Add WooCommerce Patterns Into Page › can insert WooCommerce patterns into page 

---

TimeoutError: locator.fill: Timeout 20000ms exceeded.
    Call log:
      - waiting for getByPlaceholder('Search', { exact: true })


       at ../utils/editor.js:66

      64 | const insertBlock = async ( page, blockName ) => {
      65 |      await page.getByLabel( 'Toggle block inserter' ).click();
    > 66 |      await page.getByPlaceholder( 'Search', { exact: true } ).fill( blockName );
         |                                                               ^
      67 |      await page.getByRole( 'option', { name: blockName, exact: true } ).click();
      68 |      await page.getByLabel( 'Toggle block inserter' ).click();
      69 | };




```

**Comment by adimoldovan** [comment]
> @veljkho I'm still not getting a successful run with Gutenberg. Are you running the tests locally with Gutenberg installed?
> 

It's possible that these 2 were failing because of WP6.6. Can you please check and make sure they're passing with Gutenberg with WP 6.5? The updates for WP 6.6 will be addressed in another PR.
Thanks!

**Comment by veljkho** [comment]
@adimoldovan it is [working for me](https://d.pr/i/5WoUBU) locally against WP 6.5, also I don't think the change is related to that anyway.

Might be something WP 6.6 specific.

**Comment by adimoldovan** [comment]
> @adimoldovan it is [working for me]() locally against WP 6.5, also I don't think the change is related to that anyway.
> 
> Might be something WP 6.6 specific.

Yes, it looks like WP 6.6 specific. It's being handled in https://github.com/woocommerce/woocommerce/pull/49632 and https://github.com/woocommerce/woocommerce/pull/49634. Let's hold on this one a little longer until those are fixed so we can properly validate all combinations of Gutenberg and core versions.

**Comment by adimoldovan** [comment]
@veljkho The WP 6.6. fixes are merged now, I think we can pick this one up again and see it merged. Should we give it another try?

**Comment by veljkho** [comment]
Aligning changes with the latest trunk @adimoldovan

**Comment by veljkho** [comment]
The pr is ready for merging @adimoldovan

**Comment by veljkho** [comment]
Tested locally and they are all passing with Gutenberg latest active.

**Review by adimoldovan — APPROVED** [review]
Looks good, all tests passed for me locally.
